### PR TITLE
Implement directory protection support across various web servers

### PR DESCRIPTION
**Pull request by iliaross**
This PR is related to https://github.com/virtualmin/virtualmin-nginx/pull/67.

### CODE REVIEW

File: add.cgi [hunk 1]
```diff
@@ -48,12 +48,23 @@ my $usersfile = "$dir/$htusers";
...
+		&error(&text('add_egclash', $file, $clash));
 		}
 	}
--r $usersfile && &error(&text('add_eclash2', $usersfile));
+-r $usersfile && &error(&text('add_egclash', $usersfile));
 -l $file && &error(&text('add_esymlink', $file));
 
+# Add protected directory in other webserver plugins
+foreach my $p (&virtual_server::list_feature_plugins()) {
+	my $err = &virtual_server::plugin_call($p,
```
Comment (jcameron) [inline add.cgi]: Maybe add a `plugin_defined` call to check if `feature_add_protected_dir` exists before calling it..

**Comment by jcameron** [comment]
Wow, is this all the code that's needed?

File: add.cgi (hunk 1 above)
Comment (iliaross) [reply add.cgi]: Alright, good idea. Fixed on the latest commit!

**Comment by iliaross** [comment]
> Wow, is this all the code that's needed?

Yes, it is! :--)

File: find.cgi [hunk 2]
```diff
@@ -41,19 +47,34 @@ while(my $f = <$FIND>) {
...
 			push(@dirs, [ $dir, $currfile ]);
-			print &text('find_found', "<tt>$f</tt>",
-				    "<tt>$currfile</tt>"),"<br>\n";
+			my $f_extra;
+			if (!$f_apache) {
+				# Now add newly found protected directory in other webserver plugins
+				my $currfilename = $currfile; # Extract filename from path
+				$currfilename =~ s/.*\///;
+				foreach my $p (&virtual_server::list_feature_plugins()) {
+					my ($err, $status) = &virtual_server::plugin_call($p,
```
Comment (jcameron) [inline find.cgi]: Use plugin_defined here as well

Comment (iliaross) [reply find.cgi]: Done! Thanks!

**Pull request merged**